### DISCUSSION TITLE
Seeking Guidance for Submitting Dadroit JSON Generator to Homebrew

### DISCUSSION THREAD

**Discussion by PeterVaron** [Homebrew/discussions#4926] [Writing Formulae/Casks]
### Description of issue

As the team behind Dadroit JSON Generator, an open-source CLI tool for generating JSON from schema templates, we're exploring the best path for Homebrew submission. Our tool, hosted on [GitHub](https://github.com/DadroitOrganization/Generator) and popular as a VSCode extension with about 20,000 installs ([VSCode Marketplace Link](https://marketplace.visualstudio.com/items?itemName=Dadroit.dadroit-json-generator)), is available both as source code and pre-built binaries.

We're contemplating Homebrew Core or Cask submissions but are mindful of the installation complexities. Building from source involves dependencies on Lazarus and Freepascal, leading to significant download requirements for users. To streamline the user experience, we're considering distributing pre-built binaries instead.

Our queries are:

1. Does Homebrew Core accept CLI tools' pre-built binaries?
2. If Core isn't suitable, could Homebrew Cask be an alternative for a CLI app?
3. What's the recommended approach to ensure easy accessibility for the community? Are there other suitable Homebrew channels?

Thank you for guiding us in making Dadroit JSON Generator more accessible via Homebrew.

**Comment by SMillerDev** [comment]
> Building from source involves dependencies on Lazarus and Freepascal, leading to significant download requirements for users. 

Users in the supported configuration never build from source and get a Homebrew CI build binary instead.

> Does Homebrew Core accept CLI tools' pre-built binaries?

No: https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae
> If Core isn't suitable, could Homebrew Cask be an alternative for a CLI app?

No: https://docs.brew.sh/Acceptable-Casks#rejected-casks
> What's the recommended approach to ensure easy accessibility for the community? Are there other suitable Homebrew channels?

Homebrew/core since your concern does not actually apply.

**Comment by alebcay** [comment]
In addition to Sean's response regarding the three questions you asked, I'll also propose yet another alternative approach, if you're not satisfied with the answers.

Anyone can host their own [tap](https://docs.brew.sh/Taps), which in Homebrew parlance is a third-party repository (somewhat analogous to a PPA in Ubuntu, a Debian repository, or sort of a more decentralized version of AUR for Arch). End users can add your tap to be able to install software from it.

---

Pros:

- **Not beholden to the acceptable formulae/casks criteria that homebrew-core or homebrew-cask are subject to.** You can ship your prebuilt binaries as formulae if you wish; you could ship your CLI app as a Cask if you would prefer.
- **Your software can use any license you wish.** Non-FOSS software is accepted in homebrew-cask, but only software made available under an OSI/DFSG-compatible license is accepted into homebrew-core. IANAL, but I think your modifications to the GPLv3 would likely cause your software not to be acceptable in homebrew-core (just my opinion as one maintainer, don't let that stop you from opening a PR at least for discussion/to get some more maintainers' eyes on it). If you host your own tap, these criteria are not relevant.
- **You fully own and control the formula/cask for your software.** Formulae/casks submitted into homebrew-core/homebrew-cask are open to contributions from the community, and acceptance of these contributions is up to Homebrew maintainers. We generally operate on an "open maintainer" model, where no single maintainer is responsible for a particular formula/cask. It would generally not be acceptable for a software vendor/upstream to require that all changes to their software's formula/cask in homebrew-core/homebrew-cask go through them before acceptance.

Cons:

- **Not as discoverable.** Users must tap your repo (e.g. `brew tap DadroitOrganization/tap`) before being able to install or find your software.
- **You are fully responsible for the continual maintenance and upkeep of your formula/cask.** This includes any necessary changes to the formula/cask file itself as well as any CI infrastructure you may want to have to serve prebuilt binaries to users (or perhaps, if you're already planning to provide prebuilt binaries, this is not as much of a concern).
